clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2638,6 +2638,28 @@ def SinOp : UnaryFPToFPBuiltinOp<"sin">;
 def SqrtOp : UnaryFPToFPBuiltinOp<"sqrt">;
 def TruncOp : UnaryFPToFPBuiltinOp<"trunc">;
 
+//===----------------------------------------------------------------------===//
+// Expect op
+//===----------------------------------------------------------------------===//
+
+def ExpectOp : CIR_Op<"expect", [Pure, AllTypesMatch<["result", "val", "expected"]>]> {
+  let summary = "Provides information about expected value of val";
```

Comment:
I believe we can improve this a bit: `"Compute whether expression is likely to evaluate to a specified value";`